internal/eval/eval.go

```diff
@@ -3522,6 +3528,150 @@ func evalJSONNUMINCRBY(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(resultString, false)
 }
 
+
+func evalDUMP(args []string, store *dstore.Store) []byte {
+    if len(args) < 1 {
+        return diceerrors.NewErrArity("DUMP")
+    }
+    key := args[0]
+    obj := store.Get(key)
+    if obj == nil {
+        return diceerrors.NewErrWithFormattedMessage("-ERR could not perform this operation on a key that doesn't exist")
+    }
+	
+    serializedValue, err := rdbSerialize(obj)
+    if err != nil {
+        return diceerrors.NewErrWithMessage("serialization failed")
+    }
+	encodedResult := base64.StdEncoding.EncodeToString(serializedValue)
+    return clientio.Encode(encodedResult, false)
+}
+
+func evalRestore(args []string, store *dstore.Store) []byte {
+	if len(args) < 3 {
+		return diceerrors.NewErrArity("RESTORE")
+	}
+
+	key := args[0]
+	ttlStr:=args[1]
+	ttl, _ := strconv.ParseInt(ttlStr, 10, 64)
+	
+	encodedValue := args[2]
+	serializedData, err := base64.StdEncoding.DecodeString(encodedValue)
+	if err != nil {
+		return diceerrors.NewErrWithMessage("failed to decode base64 value")
+	}
+
+	obj, err := rdbDeserialize(serializedData)
+	if err != nil {
+		return diceerrors.NewErrWithMessage("deserialization failed: " + err.Error())
+	}
+
+	newobj:=store.NewObj(obj.Value,ttl,obj.TypeEncoding,obj.TypeEncoding)
+	var keepttl=true
+
+	if(ttl>0){
+		store.Put(key, newobj, dstore.WithKeepTTL(keepttl))
+	}else{
+		store.Put(key,obj)
+	}
+	
+	return clientio.RespOK
+}
+
+func rdbDeserialize(data []byte) (*object.Obj, error) {
+	if len(data) < 3 {
+		return nil, errors.New("insufficient data for deserialization")
+	}
+	objType := data[1] 
+	switch objType {
+	case 0x00:
+		return readString(data[2:])
+	case 0xC0: // Integer type
```

Comment:
any particular reason 0x00 and 0xC0 were chosen? I have not delved much into redis implementation so I am not sure about this.

Reply:
I searched about this and according to Redis standards first two byte values determine the type. 0x00 is considered string and 0xc0 is of integer.